bip-0000.mediawiki

```diff
@@ -0,0 +1,335 @@
+* The sender should sign with one of the sighash flags ''ALL, SINGLE, NONE'' (''ANYONECANPAY'' is unsafe). It is strongly recommended implementations only use ''SIGHASH_ALL'' for silent payments<ref name="why_sighash_all">'''Why recommend ''SIGHASH_ALL''?''' Since the output address for the receiver is derived from from the sum of the ''[[#inputs-for-shared-secret-derivation|Inputs For Shared Secret Derivation]]'' public keys, the inputs must not change once the sender has signed the transaction. If the inputs are allowed to change after the fact, the receiver will not be able to calculate the shared secret needed to find and spend the output. It is currently an open question on how a future version of silent payments could be made to work with new sighash flags such as ''SIGHASH_GROUP'' and ''SIGHASH_ANYPREVOUT''.</ref>
+* Inputs used to derive the shared secret are from the ''[[#inputs-for-shared-secret-derivation|Inputs For Shared Secret Derivation]]'' list
+
+A transaction is not a ''Silent Payments v0'' transaction and should be skipped entirely<ref name="why_skip_transactions">'''Why skip transactions that spend unknown output scripts?''' Skipping transactions that spend unknown output scripts allows us to have a clean upgrade path for Silent Payments by avoiding the need to scan the same transaction multiple times with different rule sets. If a fancy new output type is added in the future and Silent Payments v1 is released with support, we would want to avoid having to first scan the transaction with the silent payment v0 rules and then again with the silent payment v1 rules.</ref> when scanning if any of the following are true:
+
+* The transaction does not contain any BIP341 taproot outputs
+* The transaction does not have at least one input from the ''[[#inputs-for-shared-secret-derivation|Inputs For Shared Secret Derivation]]'' list
+* The transaction inputs contain a new, undefined output type (e.g. SegWit versions > 1)
+
+=== Address encoding ===
+
+A silent payment address is constructed in the following manner:
+
+* Let ''B<sub>scan</sub>, b<sub>scan</sub> = Receiver's [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki BIP340] scan public key and corresponding private key''
+* Let ''B<sub>spend</sub>, b<sub>spend</sub> = Receiver's [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki BIP340] spend public key and corresponding private key''
+* Let ''B<sub>m</sub> = B<sub>spend</sub> + m·G'', where ''m'' an optional integer tweak for labeling
+** In the case of ''m'' = 0, no label is applied and ''B<sub>m</sub> = B<sub>spend</sub>''
+* The final address is a [https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki Bech32m] encoding of:
+** The human-readable part "sp" for mainnet, "tsp" for testnets (e.g.  signet, testnet, regtest)
```

Comment:
In the implementation PR the human-readable part for regtest is "sprt" but here it is "tsp", which one is correct?

Reply:
The implementation is out of date, will fix today. 

On this topic I've gotten feedback both ways: "regtest should be sprt to be consistent with bitcoin core" and "a special hrp for regtest is unnecessary." I'm leaning towards keeping it as-is in the BIP and using one HRP for all test networks, but curious what your opinion on this is?

Reply:
Did some digging on this and have come to the following conclusion:

* `regtest` is a bitcoin core implementation specific concept. As such, I don't think it should be specified in a BIP
* its fine for bitcoin core to use the regtest HRP (for consistency with our own internal testing framework), but this could just as easily be swapped out for `tsp` or any other random string for regtest

Based on this, I think its best to remove any reference to regtest from the BIP, and also keep regtest handling consistent with how other addresses are treated in bitcoin core.

For more context, see: https://github.com/bitcoin/bitcoin/issues/12314

Reply:
I'm leaning slightly towards "sprt" for regtest, but I guess it doesn't really need to be specified. BIP173 doesn't specify the regtest HRP either.